### PR TITLE
Grab all repositories, not just the first 100

### LINKED ISSUE #52: "Add a new deploy" only shows 100 repos, so some are missing
The problem is on [this line](https://github.com/hudl/black-mesa/blob/master/WebApp/WebApp/Controllers/Api/GithubController.cs#L26). To fix it, Black Mesa should query all pages of repos until there are none left. 

It would be helpful if the repositories that will never be used could be filtered out.

### PR DESCRIPTION
**Pull request by samervin**
[DEV-736](http://jira.hudl.com/browse/DEV-736)

Fixes #52. An updated version of #54, broken into a new PR for simplicity's sake.

This solution feels way better. Feedback still welcome.

### CODE REVIEW

File: WebApp/WebApp/Controllers/Api/GithubController.cs [hunk 1]
```diff
@@ -16,15 +16,41 @@
...
+        {
+            int pageSize = 100;
+            int currentPage = 1;
+
+            var allRepos = await GetRepositoryPage(pageSize, currentPage);
+
+            // GitHub does pass the next page back in the headers,
+            // but this code was easier to write and will behave in any sane use case.
+
+            while (allRepos.Count % pageSize == 0)
```
Comment (llaughlin) [inline WebApp/WebApp/Controllers/Api/GithubController.cs]: :+1: I wouldn't have though of that, neat way to check if there are more pages!

File: WebApp/WebApp/Controllers/Api/GithubController.cs [hunk 2]
```diff
@@ -16,15 +16,41 @@
 
 namespace WebApp.Controllers.Api
 {
+    class GitHubRepo
+    {
+        [JsonProperty(PropertyName = "name")]
+        public string name { get; set; }
```
Comment (tysonstewart) [inline WebApp/WebApp/Controllers/Api/GithubController.cs]: C# coding standards prefer that `name` be proper case: `Name`.

**Comment by llaughlin** [comment]
Excellent work, very clean!

File: WebApp/WebApp/Controllers/Api/GithubController.cs (hunk 1 above)
Comment (tysonstewart) [inline WebApp/WebApp/Controllers/Api/GithubController.cs]: I think there might be an edge case here with an infinite loop where there are, say, exactly 200 repos.

**Pull request merged**